ryhope/src/error.rs

```diff
@@ -0,0 +1,66 @@
+use thiserror::Error;
+use tokio_postgres::error::Error as PgError;
+
+#[derive(Error, Debug)]
+pub enum RyhopeError {
+    /// An error that occured while interacting with the DB.
+    #[error("DB error while {msg}: {err}")]
+    DbError { msg: String, err: PgError },
+
+    /// An error that occured while interacting with the DB.
+    #[error("DB error while {msg}: {err}")]
+    DbPoolError {
+        msg: String,
+        err: bb8::RunError<PgError>,
+    },
+
+    /// The internal state is incoherent; this is a bug.
+    #[error("internal Error: {0}")]
+    Internal(String),
+
+    /// Unable to extract data from the DB
+    #[error("unable to deserialize data while {msg}: {err}")]
+    InvalidFormat { msg: String, err: PgError },
```

Comment:
Same here - what actions should be taken, does this means the DB has been corrupted or someone inserted stg in the db without passing through ryhope, right ?

Reply:
Yes, or that we updated the storage format in a non retrocompatible way; in any case that's a serious issue that should ideally never happen.

Reply:
Can you write this in the error message please ?